### ISSUE TITLE
after editing name, the old slug errors

### ISSUE THREAD

**Issue by mhurwi** [OpenHunting/openhunt#89]
example: I fixed 'newsletterstah' to be 'newslettersstash'

the link in the audit log points to 'newsletterstah', and navigating to that link gets an application error.

2 possibilities to fix this: 

1- audit log should provide working link to new slug
2- keep history of old slugs, and redirect them to the new slugs

**Comment by rmcfadzean** [comment]
I feel like the audit system is the part that needs the rework the most. The 'audited object' should be some kind of polymorphic association. Hard linking to things is just going to cause headaches now and down the line. The hard linking is kind of systemic of the way the routing's been done in this project though (manually setting each route instead of RESTy/resource-y style). Which isn't to say the current style is "wrong", it just makes things more somewhat more complicated once you're (somewhat needlessly) working outside 'the rails way'.

See things like:

``` ruby
# routes.rb
resources :projects, only: [:index, :new, :show, :create, :edit] do
  member do
    post 'upvote'
  end
end

# models/project.rb
def to_param
  slug
end

# controllers/views
#   works out the controller (ProjectsController#show),
#   finds it's path ('/projects/:id)
#   gets param from @project.to_param 
url_for @project
# => /projects/example-project
project_path(@project)
# => /projects/example-project
link_to @project.name, @project
# => <a href="/projects/example-project">Example Project</a>
link_to @project.name, upvote_project_path(@project), method: :post
# => <a href="/projects/example-project" data-method="post">Example Project</a>
```

**Comment by jacquescrocker** [comment]
@wonderbread I agreed about the audit model not being super thought through. 

Personally, I'm not a fan of the way rails resources abstracts away a very simple route mapping file (for the sake of brevity). 

I guess it's a matter of taste, but I prefer an explicit [routing file](https://github.com/OpenHunting/openhunt/blob/master/config/routes.rb), where I can see exactly where each path goes. 

I also try to avoid using built in url helpers in rails. As more functionality moves to JS, I found it's much simpler just making your routes a static part of the API, so you can reference them without worrying about them changing (which web apps shouldn't ever do anyways).

**Comment by jacquescrocker** [comment]
i think the real bug here is that `set_slug` should only be set on `before_create` not `before_save`

**Comment by jacquescrocker** [comment]
fixed via 98ef0153850e50dc0e862c9677c56cdd59e865b6
